test/ex_machina/ecto_test.exs

```diff
@@ -44,18 +45,38 @@ defmodule ExMachina.EctoTest do
     }
   end
 
-  test "params_for/2 raises when passed a map" do
-    assert_raise ArgumentError, fn ->
-      TestFactory.params_for(:user_map)
-    end
-  end
-
   test "params_for/2 removes fields with nil values" do
     assert TestFactory.params_for(:user, admin: nil) == %{
-      name: "John Doe"
+      name: "John Doe",
+      articles: [],
     }
   end
 
+  test "params_for/2 recursively strips has_many values" do
+    article = TestFactory.build(:article)
+
+    user_params = TestFactory.params_for(:user, articles: [article])
+
+    assert user_params[:articles] == [%{
```

Comment:
I appreciate that in these assertions you're trying to express this _nested capabilities_ of what's produced by the factories (in the end, this is what's added in this PR), but would you consider describing _the whole structure_? Something like:

```elixir
assert user_params == %{
  name: "John Doe",
  admin: false,
  articles: [
    %{
      title: "My Awesome Article"
    }
  ],
}
```

could serve documentation purpose in longer term?

Just a thought... 💭